src/components/ProductList/ProductList.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import {
+  AsideBar,
+  AsideItem,
+  ImageGrid,
+  ImageResult,
+  ProductListMain, 
+  Spiner} from './ProductList.styles';
+import CategoriesData from '../../mocks/en-us/product-categories.json';
+import FeaturedProducts from '../../mocks/en-us/featured-products.json';
+import { ImageWrap, InfoImage } from '../Home/FeaturedProducts/FeaturedProducts.styled';
+import { PaginationControl } from './PaginationControl/PaginationControl';
+
+export const ProductList = () => {
+  const [results, setResults] = useState([]);
+  const [filters, setFilters] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  let filtersSet = [];
+
+  useEffect(() => {
+    setIsLoading(true);
+    setTimeout(() => {
```

Comment:
always that the code has a settimeout or setinterval, clear them. https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout

to avoid memory leaks

Reply:
Completely agree, I fortgot that!